Examples/TodoApp/TodoApp/src/tasks/TasksViewController.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  TasksViewController.swift
+//  TodoApp
+//
+//  Created by Alexander Persian on 3/17/20.
+//  Copyright © 2020 Spotify. All rights reserved.
+//
+
+import UIKit
+
+class TasksViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+
+    @IBOutlet weak var tasksListTableView: UITableView!
+
+    private let dataSource = TaskRemoteDataSource()
+    private var addTaskModal: UIAlertController?
+    private let activityIndicator = UIActivityIndicatorView(style: .large)
+
+    private var tasks: [Task] = [] {
+        didSet { tasksListTableView.reloadData() }
+    }
+
+    // MARK: - Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupActivityIndicator()
+        setupAddTaskModal()
+        setupTableView()
+    }
+
+    // MARK: - Setup
+
+    private func setupActivityIndicator() {
+        activityIndicator.hidesWhenStopped = true
+        activityIndicator.center = self.view.center
+        self.view.addSubview(activityIndicator)
+    }
+
+    private func setupTableView() {
+        tasksListTableView.register(TasksListTableViewCell.self, forCellReuseIdentifier: "task-cell")
+        tasksListTableView.dataSource = self
+        updateTaskList()
+    }
+
+    private func setupAddTaskModal() {
+        let alertController = UIAlertController(title: "New Task", message: nil, preferredStyle: .alert)
+        alertController.addTextField { textField in
+            textField.placeholder = "Title"
+        }
+        alertController.addTextField { textField in
+            textField.placeholder = "Description"
+        }
+
+        let titleField = alertController.textFields?[0]
+        let descField = alertController.textFields?[1]
+
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        let doneAction = UIAlertAction(title: "Done", style: .default) { _ in
+            let newDetails = TaskDetails(title: titleField?.text ?? "", description: descField?.text ?? "", isCompleted: false)
+            let newTask = Task(id: UUID().uuidString, details: newDetails)
+            self.dataSource.save(task: newTask)
+            self.updateTaskList()
+        }
+        doneAction.isEnabled = false
+
+        NotificationCenter.default.addObserver(forName: UITextField.textDidChangeNotification, object: titleField, queue: .main) { _ in
+            doneAction.isEnabled = titleField?.text?.isEmpty == false
+        }
+
+        alertController.addAction(cancelAction)
+        alertController.addAction(doneAction)
+
+        addTaskModal = alertController
+    }
+
+    // MARK: - Updates
+
+    private func updateTaskList() {
+        activityIndicator.startAnimating()
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
```

Comment:
is this delay to also simulate "loading..."?

Reply:
Whoops, I think this isn't needed any longer. Will remove.